### DISCUSSION TITLE
Mac M1: install gcc/gfortran for x86 alongside gcc/gfortran for arch64?

### DISCUSSION THREAD

**Discussion by dpo** [Homebrew/discussions#3526] [Everyday usage] [answered]
Is it possible to install both side-by-side with Homebrew?

Thanks!

**Comment by carlocab** [comment, marked as answer]
The only supported way of doing this is to install a Rosetta version of `brew` into `/usr/local`. You can do this by installing Rosetta and then doing something like
```sh
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Reply by dpo** [reply]
That sounds good, thanks! Is there a supported way for one to switch between x86_64 brew and arch64 brew?

**Reply by SMillerDev** [reply]
Yes, just call one or the other. `/usr/local/bin/brew` or `/opt/homebrew/bin/brew`

**Reply by dpo** [reply]
Excellent, thank you!

**Reply by carlocab** [reply]
Don't forget to run `/usr/local/bin/brew` in Rosetta, and don't run `/opt/homebrew/bin/brew` in Rosetta.

You might wish to set
```sh
alias brew='arch -arm64 /opt/homebrew/bin/brew'
alias ibrew='arch -x86_64 /usr/local/bin/brew'
```
or similar to avoid relying on the state of your terminal process.
